sockeye/output_handler.py

```diff
@@ -393,32 +393,26 @@ def handle(self,
     def reports_score(self) -> bool:
         return False
 
-
-class NBestOutputHandler(OutputHandler):
+class JSONOutputHandler(OutputHandler):
     """
-    Output handler to output nbest translations together with alignments,
-    as JSON.
-
-    :param stream: Stream to write translations and alignments to.
-    :param threshold: Threshold for including alignment links.
+    Output single-line JSON objects.
+    Carries over extra fields from the input.
     """
-
-    def __init__(self, stream, threshold: float) -> None:
+    def __init__(self, stream, threshold: float = 0.0) -> None:
         self.stream = stream
-        self.threshold = threshold
+        self.align_threshold = threshold
 
     def handle(self,
                t_input: inference.TranslatorInput,
                t_output: inference.TranslatorOutput,
                t_walltime: float = 0.):
 
-        extracted_alignments = []
-        for alignment_matrix in t_output.nbest_attention_matrices:
-            extracted_alignments.append(list(get_alignments(alignment_matrix, threshold=self.threshold)))
-
-        d_ = {"translations": t_output.nbest_translations,
-              "scores": t_output.nbest_scores,
-              "alignments": extracted_alignments}
+        d_ = t_output.json(self.align_threshold)
+        # copy over keys from the input that aren't present in the output
+        if t_input.raw_dict is not None:
```

Comment:
as said above, I think this logic should be part of TranslatorOutput. What do you think?

Reply:
I think this behavior is a function of the output handler: TranslatorOutput assembles all the fields that Sockeye cares about, and then the output handler takes care of merging them. But I don't feel strongly about it, and mostly I did it this way because TranslatorOutput objects don't have access to the TranslatorInput, so we'd have to add that to its constructor. I'm happy to do that and move it over if you prefer!

Reply:
Couldn't we add a "pass_through_dict" argument to the constructor of TranslatorOutput (just like your doc string change suggests), and at construction time of the object (https://github.com/awslabs/sockeye/blob/master/sockeye/inference.py#L1499) we pass in `translator_input.pass_through_dict` ? Basically, we make it the responsibility of the `_make_result()` method to carry over this additional data from input to output.

Reply:
Sure, sounds good.